spec/controllers/comments_controller_spec.rb

```diff
@@ -0,0 +1,247 @@
+      get :hide_comments, comment_id: unreviewed_comment.id
+      expect(flash[:error]).to be_nil
+      expect(response).to redirect_to(comment_path(unreviewed_comment, anchor: 'comments'))
+    end
+  end
+
+  describe "GET #add_comment" do
+    context "when comment is unreviewed" do
+      it "redirects to the comment path with add_comment params and without an error" do
+        get :add_comment, comment_id: unreviewed_comment.id
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(comment_path(unreviewed_comment, add_comment: true, anchor: 'comments'))
+      end
+    end
+  end
+
+  describe "GET #cancel_comment" do
+    context "with only valid params" do
+      it "redirects to comment path with the comments anchor and without an error" do
+        get :cancel_comment, comment_id: comment.id
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(comment_path(comment, anchor: "comments"))
+      end
+    end
+
+    context "with valid and invalid params" do
+      it "removes invalid params and redirects without an error to comment path with valid params and the comments anchor" do
+        get :cancel_comment, comment_id: comment.id, show_comments: 'yes', random_option: 'no'
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(comment_path(comment, show_comments: 'yes', anchor: "comments"))
+      end
+    end
+  end
+
+  describe "GET #cancel_comment_reply" do
+    context "with only valid params" do
+      it "redirects to comment path with the comments anchor and without an error" do
+        get :cancel_comment_reply, comment_id: comment.id
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(comment_path(comment, anchor: "comments"))
+      end
+    end
+
+    context "with valid and invalid params" do
+      it "removes invalid params and redirects without an error to comment path with valid params and the comments anchor" do
+        get :cancel_comment_reply, comment_id: comment.id, show_comments: 'yes', random_option: 'no'
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(comment_path(comment, show_comments: 'yes', anchor: "comments"))
+      end
+    end
+  end
+
+  describe "GET #cancel_comment_delete" do
+    it "redirects to the comment on the commentable without an error" do
+      get :cancel_comment_delete, id: comment.id
+      expect(flash[:error]).to be_nil
+      expect(response).to redirect_to(work_path(comment.ultimate_parent, show_comments: true, anchor: "comment_#{comment.id}"))
+    end
+  end
+
+  describe "GET #cancel_comment_edit" do
+    it "redirects to the comment on the commentable without an error" do
+      get :cancel_comment_edit, id: comment.id
+      expect(flash[:error]).to be_nil
+      expect(response).to redirect_to(work_path(comment.ultimate_parent, show_comments: true, anchor: "comment_#{comment.id}"))
+    end
+  end
+
+  describe "GET #destroy" do
```

Comment:
Could we have another test where `@comment.destroy_or_mark_deleted` is stubbed to return false, so we can hit lines 301-302?